dvc/repo/experiments/remove.py

```diff
@@ -5,21 +5,25 @@
 from dvc.repo.scm_context import scm_context
 
 from .base import EXPS_NAMESPACE, ExpRefInfo
-from .utils import exp_refs_by_name, remove_exp_refs
+from .utils import exp_refs, exp_refs_by_name, remove_exp_refs
 
 logger = logging.getLogger(__name__)
 
 
 @locked
 @scm_context
-def remove(repo, exp_names=None, queue=False, **kwargs):
-    if not exp_names and not queue:
+def remove(repo, exp_names=None, queue=False, workspace=False, **kwargs):
+    if not any([exp_names, queue, workspace]):
         return 0
 
     removed = 0
-    if queue:
+    if queue or workspace:
         removed += len(repo.experiments.stash)
         repo.experiments.stash.clear()
+    if workspace:
+        ref_infos = list(exp_refs(repo.scm))
+        remove_exp_refs(repo.scm, ref_infos)
```

Comment:
This will remove all experiments in the user's entire git repository. `exp remove --workspace` should only remove experiments that are based on the user's current `HEAD` (and not experiments based on other git commits)

https://github.com/iterative/dvc/issues/5676#issuecomment-858582527

so we can't use the general `exp_refs()` iterator here, it should be:
```
        head = repo.scm.get_ref()
        exp_refs_by_baseline(head)
```

Reply:
I think it's fine to leave the behavior as is and rename the flag to `--all`. That fits the requirement here and is easier to explain.

Reply:
@karajan1001 can you please rename the flag to `--all`? If it is left as `--workspace` it will be confusing, since all other DVC commands use `--workspace` to mean `HEAD` + whatever is in the actual working directory

Reply:
@pmrowla , @dberenbaum 
I think the feature `--all` is not as useful as `--workspace`?  Can add it sometime later?

Reply:
I think which is more useful depends. Most of the time, I would guess users won't notice a different between them. If they have experiments in other commits, they might want to keep those, but I think it's just as likely that they have forgotten about them or don't know they still exist and would be happy to drop them.

After the discussion in this ticket, I prefer `--all` because:

1. It's easier to explain and understand. It's not obvious what is a "workspace" experiment, and users might want to confirm that these are the experiments they want to remove.
2. It parallels `dvc exp list`. If we eventually want to do something like `dvc exp remove --remote`, having this parallel `dvc exp list` seems useful since it's easy to see what experiments will be deleted.
3. Similarly, in the future we could add the `--rev` option like in `dvc exp list`. This is probably more useful than the `--workspace/--all-branches/--all-tags/--all-commits` flags since `--rev` is more granular (and `--rev HEAD` can do the same as `--workspace`), and I don't see `--all-branches` or `--all-tags` ever being useful in `dvc exp remove`.